guides/micronaut-data-jdbc-repository/java/src/main/java/example/micronaut/GenreController.java

```diff
@@ -0,0 +1,90 @@
+package example.micronaut;
+
+import example.micronaut.domain.Genre;
+import io.micronaut.data.model.Pageable;
+import io.micronaut.data.model.Sort;
+import io.micronaut.http.HttpHeaders;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.annotation.Body;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Delete;
+import io.micronaut.http.annotation.Get;
+import io.micronaut.http.annotation.Post;
+import io.micronaut.http.annotation.Put;
+import io.micronaut.http.annotation.Status;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.scheduling.TaskExecutors;
+import io.micronaut.scheduling.annotation.ExecuteOn;
+import io.micronaut.data.exceptions.DataAccessException;
+import javax.validation.Valid;
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@ExecuteOn(TaskExecutors.IO)  // <1>
+@Controller("/genres")  // <2>
+public class GenreController {
+
+    protected final GenreRepository genreRepository;
+
+    public GenreController(GenreRepository genreRepository) { // <3>
+        this.genreRepository = genreRepository;
+    }
+
+    @Get("/{id}") // <4>
+    public Optional<Genre> show(Long id) {
+        return genreRepository
+                .findById(id); // <5>
+    }
+
+    @Put // <6>
+    public HttpResponse update(@Body @Valid GenreUpdateCommand command) { // <7>
+        int numberOfEntitiesUpdated = genreRepository.update(command.getId(), command.getName());
```

Comment:
The value of `numberOfEntitiesUpdated` is not used. Maybe you can remove the variable and call the method in the repository.

Reply:
I've removed the unused variable